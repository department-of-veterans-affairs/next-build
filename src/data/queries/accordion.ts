import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAccordion } from '@/types/drupal/paragraph'
import { Accordion } from '@/types/formatted/accordion'

export const formatter: QueryFormatter<ParagraphAccordion, Accordion> = (
  entity: ParagraphAccordion
) => {
  return {
    id: entity.id,
    header: entity.field_header || null,
    html: entity.field_rich_wysiwyg.processed,
  }
}
