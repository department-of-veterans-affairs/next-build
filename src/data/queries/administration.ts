import { FieldAdministration } from '@/types/drupal/field_type'

export interface Administration {
  entityId: number
  name: string | null
}

/**
 * Formats a Drupal administration field into our standardized Administration interface.
 * 
 * In our Drupal content, the `field_administration` field points to a taxonomy Term
 * that is currently called "Section", but "Section" is only a human label added to
 * it in the Drupal interface, and that name has evolved over time. It was originally,
 * "administration" (and thus the name of the field). Since the only machine name of
 * it is `field_administration`, we'll continue to call it "administration" in our code.
 */
export const formatAdministration = (
  administration: FieldAdministration | null | undefined
): Administration => ({
  entityId: administration?.drupal_internal__tid || null, // "t" is for taxonomy term
  name: administration?.name || null,
})
