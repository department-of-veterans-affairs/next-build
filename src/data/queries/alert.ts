// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter } from 'next-drupal-query'
import { Alert } from '@/types/formatted/alert'

export const formatter: QueryFormatter<BlockAlert, Alert> = (
  entity: BlockAlert
) => {
  return {
    alertType:
      entity.field_alert_type === 'information'
        ? 'info'
        : entity.field_alert_type,
    id: entity.id,
    title: entity.field_alert_title,
    content: {
      header: entity.field_alert_content.field_text_expander,
      text: entity.field_alert_content.field_wysiwyg.processed,
    },
  }
}
