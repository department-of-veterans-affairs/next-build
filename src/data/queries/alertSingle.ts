import { QueryFormatter } from 'next-drupal-query'
import { ParagraphAlertSingle } from '@/types/drupal/paragraph'
import {
  AlertSingle,
  AlertBlock,
  AlertNonReusable,
} from '@/types/formatted/alert'
import { formatParagraph } from '@/lib/drupal/paragraphs'
import { queries } from '.'

// TODO:
//
// EDIT:
// It seems *maybe* that next-drupal-query is designed to handle this already.
// You can pass an `id` to queries.getParams().
// E.g.
// queries.getParams('paragraph--alert_single'))
//
// We don't need this here:
// export const params: QueryParams<null> = () => {
//   return queries
//     .getParams()
//     .addInclude([
//       'field_alert_block_reference',
//       'field_alert_block_reference.field_alert_content',
//       'field_alert_non_reusable_ref',
//       'field_alert_non_reusable_ref.field_va_paragraphs',
//     ])
// }
//
// ...but it seems like we might want to define
// paragraph-specific `includes` within the paragraph's
// query file (this file), and then somehow reference these in
// the node query when we need to include fields of a specific
// paragraph type.
//
// E.g.
// export const include = [
//   'field_alert_block_reference',
//   'field_alert_block_reference.field_alert_content',
//   'field_alert_non_reusable_ref',
//   'field_alert_non_reusable_ref.field_va_paragraphs',
// ]
//
// /* Some util file*/
// export const getIncludedSubFields = (
//   parentField: string,
//   subFields: string[]
// ) => [parentField, ...subFields.map((subField) => `${parentField}.${subField}`)]
//
// /* src/data/queries/resourcesSupport.ts */
// import { include } from '@/data/queries/alertSingle'
// export const params: QueryParams<null> = () => {
//   return queries.getParams().addInclude([
//     ...getIncludedSubFields('field_alert_single', include)
//   ])
// }

export const formatter: QueryFormatter<ParagraphAlertSingle, AlertSingle> = (
  entity: ParagraphAlertSingle
) => {
  return {
    type: entity.type as AlertSingle['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertSelection:
      entity.field_alert_selection as AlertSingle['alertSelection'],
    blockReference: queries.formatData(
      'block--alert',
      entity.field_alert_block_reference
    ) as AlertBlock,
    nonReusableRef: formatParagraph(
      entity.field_alert_non_reusable_ref
    ) as AlertNonReusable,
  }
}
