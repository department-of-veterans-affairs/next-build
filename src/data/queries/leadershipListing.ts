import { QueryData, QueryFormatter, QueryParams } from 'next-drupal-query'
import { DrupalJsonApiParams } from 'drupal-jsonapi-params'
import { NodeLeadershipListing } from '@/types/drupal/node'
import { LeadershipListing } from '@/types/formatted/leadershipListing'
import { RESOURCE_TYPES } from '@/lib/constants/resourceTypes'
import { ExpandedStaticPropsContext } from '@/lib/drupal/staticProps'
import { buildSideNavDataFromMenu } from '@/lib/drupal/facilitySideNav'
import { queries } from '.'
import {
  entityBaseFields,
  fetchSingleEntityOrPreview,
  getMenu,
} from '@/lib/drupal/query'
import { Menu } from '@/types/drupal/menu'

// Define the query params for fetching node--leadership_listing.
export const params: QueryParams<null> = () => {
  return new DrupalJsonApiParams().addInclude([
    'field_leadership',
    'field_office',
    'field_leadership.field_media.image',
    'field_leadership.field_telephone',
  ])
}

// Define the option types for the data loader.
export type LeadershipListingDataOpts = {
  id: string
  context?: ExpandedStaticPropsContext
}

export type LeadershipListingData = {
  entity: NodeLeadershipListing
  menu: Menu | null
}

// Implement the data loader.
export const data: QueryData<
  LeadershipListingDataOpts,
  LeadershipListingData
> = async (opts) => {
  const entity = (await fetchSingleEntityOrPreview(
    opts,
    RESOURCE_TYPES.LEADERSHIP_LISTING,
    params
  )) as NodeLeadershipListing

  const menu = entity.field_office
    ? await getMenu(
        entity.field_office.field_system_menu.resourceIdObjMeta
          .drupal_internal__target_id
      )
    : null

  return { entity, menu }
}

export const formatter: QueryFormatter<
  LeadershipListingData,
  LeadershipListing
> = ({ entity, menu }) => {
  const formattedMenu =
    menu !== null ? buildSideNavDataFromMenu(entity.path.alias, menu) : null
  const formattedProfiles = entity.field_leadership.map((profile) => {
    return {
      firstName: profile.field_name_first,
      lastName: profile.field_last_name,
      vamcTitle: profile.field_office?.title || '',
      description: profile.field_description || '',
      suffix: profile.field_suffix || '',
      phoneNumber: queries.formatData(
        'paragraph--phone_number',
        profile.field_telephone
      ),
      media: queries.formatData('media--image', profile.field_media),
      link: profile.path.alias,
      id: profile.id,
    }
  })
  return {
    ...entityBaseFields(entity),
    introText: entity.field_intro_text,
    profiles: formattedProfiles,
    menu: formattedMenu,
  }
}
