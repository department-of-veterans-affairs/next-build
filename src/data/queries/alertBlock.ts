// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter } from 'next-drupal-query'
import { AlertBlock, AlertType } from '@/types/formatted/alert'

export const formatter: QueryFormatter<BlockAlert, AlertBlock> = (
  entity: BlockAlert
) => {
  if (!entity) {
    return null
  }

  return {
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    id: entity.id,
    title: entity.field_alert_title,
    content: {
      header:
        entity.field_alert_content.field_text_expander ??
        entity.field_alert_heading,
      text: entity.field_alert_content?.field_wysiwyg?.processed ?? '',
    },
  }
}
