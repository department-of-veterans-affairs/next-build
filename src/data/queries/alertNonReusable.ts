// Define the query params for fetching block--alert.
import { BlockAlert } from '@/types/drupal/block'
import { QueryFormatter } from 'next-drupal-query'
import { AlertType, AlertNonReusable } from '@/types/formatted/alert'
import { ParagraphNonReusableAlert } from '@/types/drupal/paragraph'
import { formatParagraph } from '@/lib/drupal/paragraphs'

export const formatter: QueryFormatter<
  ParagraphNonReusableAlert,
  AlertNonReusable
> = (entity: ParagraphNonReusableAlert) => {
  if (!entity) {
    return null
  }

  return {
    type: entity.type as AlertNonReusable['type'],
    id: entity.id,
    entityId: entity.drupal_internal__id,
    alertType: (entity.field_alert_type === 'information'
      ? 'info'
      : entity.field_alert_type) as AlertType,
    heading: entity.field_alert_heading,
    paragraphs: entity.field_va_paragraphs.map(formatParagraph),
  }
}
