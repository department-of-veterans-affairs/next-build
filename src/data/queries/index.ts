import { createQueries } from 'next-drupal-query'
import * as NewsStory from './newsStory'
import * as NewsStoryTeaser from './newsStoryTeaser'
import * as StoryListing from './storyListing'
import * as QuestionAnswer from './questionAnswer'
import * as ExpandableText from './expandableText'
import * as LinkTeaser from './linkTeaser'
import * as MediaImage from './mediaImage'
import * as Banner from './banners'
import * as PersonProfile from './personProfile'
import * as Button from './button'
import * as AudienceTopics from './audienceTopics'
import * as Alert from './alert'
import * as EmailContact from './emailContact'
import * as BenefitsHub from './benefitsHub'
import * as Wysiwyg from './wysiwyg'
import * as StaticPathResources from './staticPathResources'
import * as HeaderFooter from './headerFooter'

export const queries = createQueries({
  // standard Drupal entity data queries
  'node--news_story': NewsStory,
  'node--news_story--teaser': NewsStoryTeaser,
  'node--story_listing': StoryListing,
  'node--q_a': QuestionAnswer,
  'node--person_profile': PersonProfile,
  'node--landing_page': BenefitsHub, // "Benefits Hub Landing Page"
  'paragraph--audience_topics': AudienceTopics,
  'paragraph--button': Button,
  'paragraph--email_contact': EmailContact,
  'paragraph--expandable_text': ExpandableText,
  'paragraph--link_teaser': LinkTeaser,
  'paragraph--rich_text_char_limit_1000': Wysiwyg,
  'paragraph--wysiwyg': Wysiwyg,
  'media--image': MediaImage,
  'block--alert': Alert,

  // Custom queries 
  'banner--alerts_lookup': Banner,
  'header-footer-data': HeaderFooter,

  // Static Path Generation
  'static-path-resources': StaticPathResources,
})
