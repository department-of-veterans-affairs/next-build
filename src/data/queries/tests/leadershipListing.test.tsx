/**
 * @jest-environment node
 */
import { NodeLeadershipListing } from '@/types/drupal/node'
import { queries } from '@/data/queries'
import mockData from '@/mocks/leadershipListing.mock.json'
import { params } from '../leadershipListing'

//eslint-disable-next-line
const leadershipListingMock: NodeLeadershipListing | any = mockData

describe('DrupalJsonApiParams configuration', () => {
  test('params function includes the correct fields', () => {
    const paramsInstance = params()
    const queryString = decodeURIComponent(paramsInstance.getQueryString())
    expect(queryString).toMatch(
      /include=field_leadership,field_office,field_leadership.field_media.image,field_leadership.field_telephone,field_leadership.field_office,field_administration/
    )
  })
})

describe('LeadershipListing formatData', () => {
  test('outputs formatted data', () => {
    expect(
      queries.formatData('node--leadership_listing', {
        entity: leadershipListingMock,
        menu: null,
      })
    ).toMatchSnapshot()
  })
  describe('Lovell variant handling', () => {
    const lovellPath = {
      alias: '/lovell-federal-health-care-va/about-us/leadership',
      pid: 79642,
      langcode: 'en',
    }
    const lovellBreadcrumbs = [
      {
        uri: 'https://va-gov-cms.ddev.site/',
        title: 'Home',
        options: [],
      },
      {
        uri: 'https://va-gov-cms.ddev.site/lovell-federal-health-care',
        title: 'Lovell Federal health care',
        options: [],
      },
      {
        uri: 'internal:#',
        title: 'ABOUT LOVELL FEDERAL',
        options: [],
      },
      {
        uri: 'https://va-gov-cms.ddev.site/lovell-federal-health-care-va/about-us',
        title: 'About us',
        options: [],
      },
      {
        uri: 'internal:#',
        title: 'Leadership',
        options: [],
      },
    ]
    test('outputs formatted data with Lovell variant', () => {
      const formattedData = queries.formatData('node--leadership_listing', {
        entity: leadershipListingMock,
        menu: null,
        lovell: {
          isLovellVariantPage: true,
          variant: 'tricare',
        },
      })
      expect(formattedData).toMatchSnapshot()
    })
    test('updates the breadcrumbs for Lovell variant', () => {
      const formattedData = queries.formatData('node--leadership_listing', {
        entity: {
          ...leadershipListingMock,
          path: lovellPath,
          breadcrumbs: lovellBreadcrumbs,
        },
        menu: null,
        lovell: {
          isLovellVariantPage: true,
          variant: 'tricare',
        },
      })
      expect(formattedData.breadcrumbs[1]).toEqual({
        uri: 'https://va-gov-cms.ddev.site/lovell-federal-health-care-tricare',
        title: 'Lovell Federal health care - TRICARE',
        options: [],
      })
    })
  })
})
