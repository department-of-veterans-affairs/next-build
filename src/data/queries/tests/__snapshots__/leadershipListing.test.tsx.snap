// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeadershipListing formatData Lovell variant handling outputs formatted data with Lovell variant 1`] = `
{
  "administration": {
    "entityId": null,
    "name": null,
  },
  "breadcrumbs": [
    {
      "options": [],
      "title": "Home",
      "uri": "https://va-gov-cms.ddev.site/",
    },
    {
      "options": [],
      "title": "VA White River Junction health care",
      "uri": "https://va-gov-cms.ddev.site/white-river-junction-health-care",
    },
    {
      "options": [],
      "title": "ABOUT VA WHITE RIVER JUNCTION",
      "uri": "internal:#",
    },
    {
      "options": [],
      "title": "About us",
      "uri": "https://va-gov-cms.ddev.site/white-river-junction-health-care/about-us",
    },
    {
      "options": [],
      "title": "Leadership",
      "uri": "internal:#",
    },
  ],
  "entityId": 27519,
  "entityPath": "/white-river-junction-health-care/about-us/leadership",
  "id": "dd02e587-f784-41cf-8489-ffad717e6a29",
  "introText": "Find contact information for members of VA White River Junction Healthcare System's leadership team.",
  "lastUpdated": "2022-02-07T21:25:26+00:00",
  "lovellSwitchPath": "/white-river-junction-health-care/about-us/leadership",
  "lovellVariant": "tricare",
  "menu": null,
  "metatags": [
    {
      "attributes": {
        "content": "Leadership | VA White River Junction health care | Veterans Affairs",
        "name": "title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Find contact information for members of VA White River Junction Healthcare System's leadership team.",
        "name": "description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "href": "https://www.va.gov/img/design/logo/va-og-image.png",
        "rel": "image_src",
      },
      "tag": "link",
    },
    {
      "attributes": {
        "content": "Veterans Affairs",
        "property": "og:site_name",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Leadership | Veterans Affairs",
        "property": "og:title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Find contact information for members of VA White River Junction Healthcare System's leadership team.",
        "property": "og:description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "https://www.va.gov/img/design/logo/va-og-image.png",
        "property": "og:image",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "U.S. Department of Veterans Affairs",
        "property": "og:image:alt",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "summary_large_image",
        "name": "twitter:card",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "@DeptVetAffairs",
        "name": "twitter:site",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Find contact information for members of VA White River Junction Healthcare System's leadership team.",
        "name": "twitter:description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Leadership | VA White River Junction health care | Veterans Affairs",
        "name": "twitter:title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "https://www.va.gov/img/design/logo/va-og-image.png",
        "name": "twitter:image",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "U.S. Department of Veterans Affairs",
        "name": "twitter:image:alt",
      },
      "tag": "meta",
    },
  ],
  "moderationState": "published",
  "profiles": [
    {
      "description": "Executive Director",
      "firstName": "Brett ",
      "id": "1d904e9d-8fa5-457d-88db-1f0e61b44c42",
      "lastName": "Rusch",
      "link": "/white-river-junction-health-care/staff-profiles/brett-rusch",
      "media": {
        "alt": "Executive Director",
        "height": 4384,
        "id": "cb77541a-6da8-4d38-93be-c84d5c595b3f",
        "links": {
          "self": {
            "href": "https://va-gov-cms.ddev.site/jsonapi/file/file/cb77541a-6da8-4d38-93be-c84d5c595b3f",
          },
        },
        "title": "",
        "type": "media--image",
        "width": 3328,
      },
      "phoneNumber": {
        "entityId": 166138,
        "extension": "5400",
        "id": "abd718dc-d96c-44ee-af54-00f6b62180ad",
        "label": null,
        "number": "802-295-9363",
        "phoneType": "phone",
        "type": "paragraph--phone_number",
      },
      "suffix": "MD",
      "vamcTitle": "VA White River Junction health care",
    },
    {
      "description": "Chief of Staff",
      "firstName": "Daniel ",
      "id": "62bdd347-52a8-4f26-95e6-d5e5a55d141e",
      "lastName": "O'Rouke",
      "link": "/white-river-junction-health-care/staff-profiles/daniel-orouke",
      "media": {
        "alt": "Chief of Staff",
        "height": 3368,
        "id": "5000eeee-6043-4d2c-83c6-937768652c2d",
        "links": {
          "self": {
            "href": "https://va-gov-cms.ddev.site/jsonapi/file/file/5000eeee-6043-4d2c-83c6-937768652c2d",
          },
        },
        "title": "",
        "type": "media--image",
        "width": 2694,
      },
      "phoneNumber": {
        "entityId": 166140,
        "extension": "5420",
        "id": "477f1bcf-b197-4d68-8a3d-f969c5af4e81",
        "label": null,
        "number": "802-295-9363",
        "phoneType": "phone",
        "type": "paragraph--phone_number",
      },
      "suffix": "Dr.",
      "vamcTitle": "VA White River Junction health care",
    },
    {
      "description": "Associate Director of Nursing and Patient Care Services ",
      "firstName": "Kelley",
      "id": "6c81991e-3bd0-469d-9ee0-816afb0ed7d1",
      "lastName": "Saindon",
      "link": "/white-river-junction-health-care/staff-profiles/kelley-saindon",
      "media": {
        "alt": "Dr. Kelley Saindon Headshot",
        "height": 3300,
        "id": "19cf0f99-1f49-46a6-8667-000778153813",
        "links": {
          "self": {
            "href": "https://va-gov-cms.ddev.site/jsonapi/file/file/19cf0f99-1f49-46a6-8667-000778153813",
          },
        },
        "title": "",
        "type": "media--image",
        "width": 2550,
      },
      "phoneNumber": null,
      "suffix": "DNP, RN, NE-BC, CHPN ",
      "vamcTitle": "VA White River Junction health care",
    },
  ],
  "published": true,
  "title": "Leadership",
  "type": "node--leadership_listing",
}
`;

exports[`LeadershipListing formatData outputs formatted data 1`] = `
{
  "administration": {
    "entityId": null,
    "name": null,
  },
  "breadcrumbs": [
    {
      "options": [],
      "title": "Home",
      "uri": "https://va-gov-cms.ddev.site/",
    },
    {
      "options": [],
      "title": "VA White River Junction health care",
      "uri": "https://va-gov-cms.ddev.site/white-river-junction-health-care",
    },
    {
      "options": [],
      "title": "ABOUT VA WHITE RIVER JUNCTION",
      "uri": "internal:#",
    },
    {
      "options": [],
      "title": "About us",
      "uri": "https://va-gov-cms.ddev.site/white-river-junction-health-care/about-us",
    },
    {
      "options": [],
      "title": "Leadership",
      "uri": "internal:#",
    },
  ],
  "entityId": 27519,
  "entityPath": "/white-river-junction-health-care/about-us/leadership",
  "id": "dd02e587-f784-41cf-8489-ffad717e6a29",
  "introText": "Find contact information for members of VA White River Junction Healthcare System's leadership team.",
  "lastUpdated": "2022-02-07T21:25:26+00:00",
  "lovellSwitchPath": null,
  "lovellVariant": null,
  "menu": null,
  "metatags": [
    {
      "attributes": {
        "content": "Leadership | VA White River Junction health care | Veterans Affairs",
        "name": "title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Find contact information for members of VA White River Junction Healthcare System's leadership team.",
        "name": "description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "href": "https://www.va.gov/img/design/logo/va-og-image.png",
        "rel": "image_src",
      },
      "tag": "link",
    },
    {
      "attributes": {
        "content": "Veterans Affairs",
        "property": "og:site_name",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Leadership | Veterans Affairs",
        "property": "og:title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Find contact information for members of VA White River Junction Healthcare System's leadership team.",
        "property": "og:description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "https://www.va.gov/img/design/logo/va-og-image.png",
        "property": "og:image",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "U.S. Department of Veterans Affairs",
        "property": "og:image:alt",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "summary_large_image",
        "name": "twitter:card",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "@DeptVetAffairs",
        "name": "twitter:site",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Find contact information for members of VA White River Junction Healthcare System's leadership team.",
        "name": "twitter:description",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "Leadership | VA White River Junction health care | Veterans Affairs",
        "name": "twitter:title",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "https://www.va.gov/img/design/logo/va-og-image.png",
        "name": "twitter:image",
      },
      "tag": "meta",
    },
    {
      "attributes": {
        "content": "U.S. Department of Veterans Affairs",
        "name": "twitter:image:alt",
      },
      "tag": "meta",
    },
  ],
  "moderationState": "published",
  "profiles": [
    {
      "description": "Executive Director",
      "firstName": "Brett ",
      "id": "1d904e9d-8fa5-457d-88db-1f0e61b44c42",
      "lastName": "Rusch",
      "link": "/white-river-junction-health-care/staff-profiles/brett-rusch",
      "media": {
        "alt": "Executive Director",
        "height": 4384,
        "id": "cb77541a-6da8-4d38-93be-c84d5c595b3f",
        "links": {
          "self": {
            "href": "https://va-gov-cms.ddev.site/jsonapi/file/file/cb77541a-6da8-4d38-93be-c84d5c595b3f",
          },
        },
        "title": "",
        "type": "media--image",
        "width": 3328,
      },
      "phoneNumber": {
        "entityId": 166138,
        "extension": "5400",
        "id": "abd718dc-d96c-44ee-af54-00f6b62180ad",
        "label": null,
        "number": "802-295-9363",
        "phoneType": "phone",
        "type": "paragraph--phone_number",
      },
      "suffix": "MD",
      "vamcTitle": "VA White River Junction health care",
    },
    {
      "description": "Chief of Staff",
      "firstName": "Daniel ",
      "id": "62bdd347-52a8-4f26-95e6-d5e5a55d141e",
      "lastName": "O'Rouke",
      "link": "/white-river-junction-health-care/staff-profiles/daniel-orouke",
      "media": {
        "alt": "Chief of Staff",
        "height": 3368,
        "id": "5000eeee-6043-4d2c-83c6-937768652c2d",
        "links": {
          "self": {
            "href": "https://va-gov-cms.ddev.site/jsonapi/file/file/5000eeee-6043-4d2c-83c6-937768652c2d",
          },
        },
        "title": "",
        "type": "media--image",
        "width": 2694,
      },
      "phoneNumber": {
        "entityId": 166140,
        "extension": "5420",
        "id": "477f1bcf-b197-4d68-8a3d-f969c5af4e81",
        "label": null,
        "number": "802-295-9363",
        "phoneType": "phone",
        "type": "paragraph--phone_number",
      },
      "suffix": "Dr.",
      "vamcTitle": "VA White River Junction health care",
    },
    {
      "description": "Associate Director of Nursing and Patient Care Services ",
      "firstName": "Kelley",
      "id": "6c81991e-3bd0-469d-9ee0-816afb0ed7d1",
      "lastName": "Saindon",
      "link": "/white-river-junction-health-care/staff-profiles/kelley-saindon",
      "media": {
        "alt": "Dr. Kelley Saindon Headshot",
        "height": 3300,
        "id": "19cf0f99-1f49-46a6-8667-000778153813",
        "links": {
          "self": {
            "href": "https://va-gov-cms.ddev.site/jsonapi/file/file/19cf0f99-1f49-46a6-8667-000778153813",
          },
        },
        "title": "",
        "type": "media--image",
        "width": 2550,
      },
      "phoneNumber": null,
      "suffix": "DNP, RN, NE-BC, CHPN ",
      "vamcTitle": "VA White River Junction health care",
    },
  ],
  "published": true,
  "title": "Leadership",
  "type": "node--leadership_listing",
}
`;
